core/core-impl/src/main/java/ai/timefold/solver/core/impl/localsearch/DefaultLocalSearchPhaseFactory.java

```diff
@@ -191,21 +199,137 @@ private UnionMoveSelectorConfig determineDefaultMoveSelectorConfig(HeuristicConf
+                            .withMoveSelectors(new ChangeMoveSelectorConfig(),
+                                    new SwapMoveSelectorConfig(),
+                                    new ChangeMoveSelectorConfig()
+                                            .withEntitySelectorConfig(new EntitySelectorConfig().withId(changeSelectorName))
+                                            .withValueSelectorConfig(new ValueSelectorConfig()
+                                                    .withNearbySelectionConfig(new NearbySelectionConfig()
+                                                            .withOriginEntitySelectorConfig(new EntitySelectorConfig()
+                                                                    .withMimicSelectorRef(changeSelectorName))
+                                                            .withNearbyDistanceMeterClass(
+                                                                    configPolicy.getNearbyDistanceMeterClass()))),
+                                    new SwapMoveSelectorConfig()
+                                            .withEntitySelectorConfig(new EntitySelectorConfig()
+                                                    .withId(swapSelectorName))
+                                            .withSecondaryEntitySelectorConfig(new EntitySelectorConfig()
+                                                    .withNearbySelectionConfig(new NearbySelectionConfig()
+                                                            .withOriginEntitySelectorConfig(new EntitySelectorConfig()
+                                                                    .withMimicSelectorRef(swapSelectorName))
+                                                            .withNearbyDistanceMeterClass(
+                                                                    configPolicy.getNearbyDistanceMeterClass()))),
+                                    new TailChainSwapMoveSelectorConfig()
+                                            .withEntitySelectorConfig(new EntitySelectorConfig()
+                                                    .withId(tailChainSelectorName))
+                                            .withValueSelectorConfig(new ValueSelectorConfig()
+                                                    .withNearbySelectionConfig(new NearbySelectionConfig()
+                                                            .withOriginEntitySelectorConfig(new EntitySelectorConfig()
+                                                                    .withMimicSelectorRef(tailChainSelectorName))
+                                                            .withNearbyDistanceMeterClass(
+                                                                    configPolicy.getNearbyDistanceMeterClass()))));
+                } else {
+                    String changeSelectorName = "changeMoveSelector-%s".formatted(UUID.randomUUID().toString().substring(0, 8));
```

Comment:
I recommend turning the code for randomness into its own method - it is repeated far too often. Something like `addRandomSuffix(...)`.
Also, any chance we could make this randomness deterministic, sharing a pre-determined random seed across all the invocations?